### ISSUE TITLE
Recommended use of `ask` is incorrect

### ISSUE THREAD

**Issue by leehambley** [capistrano/capistrano#1758]
_From @mattbrictson on May 31, 2016 19:16_

The [documentation](http://capistranorb.com/documentation/getting-started/user-input/#) recommends this pattern:

``` ruby
set :database_password, ask('Enter the database password:', 'default', echo: false)
```

But this is wrong. The first argument to `ask` is supposed to be a variable name, not a prompt string. Technically the code still works (since the outer `set` overrides the implicit `set` within `ask`), but it is very misleading.

_Copied from original issue: capistrano/documentation#192_

**Comment by InFog** [comment]
Hey there. I want to fix this one. But, as stated, the docs are really misleading.

The signature for `ask` is `ask(key, default=nil, options={})` and in the end it works "fine" because I18n's `default_question` is `question_default: "Please enter %{key} (%{default_value}): "`, so, yes, the "key" is there.

What should be the recommended way to `ask` questions with a prompt string? Or should it just be explicit that an outer `set` would to the job in such situations?

**Comment by mattbrictson** [comment]
There is no official way to `ask` a question with a custom prompt string. The technique currently in the docs is a hack, IMHO, and has no test coverage. My suggestion would be to change the docs so that it only explains these patterns:

```ruby
ask(:database_encoding, "UTF-8")
ask(:database_password, "default", echo: false)
```

If the docs also show an example of what the default prompts look like, that would make this feature is easier to understand.

As an aside, if we want to officially support custom prompts, then perhaps that could be a separate PR. It could be another supported option, like:

```ruby
ask(:database_encoding, "UTF-8", prompt: "My custom question prompt: ")
```

**Comment by mattbrictson** [comment]
Fixed in #1924
